src/sentry/integrations/slack/integration.py

```diff
@@ -120,15 +124,12 @@ def get_pipeline_views(self):
     def get_team_info(self, access_token):
         payload = {"token": access_token}
 
-        session = http.build_session()
-        resp = session.get("https://slack.com/api/team.info", params=payload)
-        resp.raise_for_status()
-        status_code = resp.status_code
-        resp = resp.json()
-        # TODO: track_response_code won't hit if we have an error status code
-        track_response_code(status_code, resp.get("ok"))
-
-        # TODO: check for resp["ok"]
+        client = SlackClient()
+        try:
+            resp = client.get("/team.info", params=payload)
+        except ApiError as e:
+            logger.error("slack.team-info.response-error", extra={"error": six.text_type(e)})
+            raise IntegrationError("Could not retrieve Slack team information.")
```

Comment:
@MeredithAnya Is there a reason you don't want to just re-raise the original exception?

Reply:
So we can catch it here: https://github.com/getsentry/sentry/blob/58da9980806d5c2c18ef6efed7f685ac75a46d11/src/sentry/integrations/pipeline.py#L42